pkg/imgpkg/cmd/copy_repo_src.go

```diff
@@ -175,6 +187,26 @@ func (o CopyRepoSrc) getSourceImages() (*ctlimgset.UnprocessedImageRefs, error)
 	panic("Unreachable")
 }
 
+func (o CopyRepoSrc) getBundleWithoutNestedBundleImagesLockLocalized(bundle *ctlbundle.Bundle) (lockconfig.ImagesLock, error) {
+	imagesLock, err := bundle.ImagesLockLocalized()
+	if err != nil {
+		if ctlbundle.IsNotBundleError(err) {
+			return lockconfig.ImagesLock{}, fmt.Errorf("Expected bundle image but found plain image (hint: Did you use -i instead of -b?)")
+		}
+		return lockconfig.ImagesLock{}, err
+	}
+	for _, i := range imagesLock.Images {
+		isBundle, err := ctlbundle.NewBundle(i.Image, o.registry).IsBundle()
+		if err != nil {
+			return lockconfig.ImagesLock{}, err
+		}
+		if isBundle {
+			return lockconfig.ImagesLock{}, fmt.Errorf("This bundle contains bundles, in order to copy please use the --experimental-recursive-bundle flag to copy nested bundles")
+		}
+	}
```

Comment:
no need to check this here since push is verifying this: https://github.com/vmware-tanzu/carvel-imgpkg/blob/f0aadec0ec3e58bf4927be4f7ad87bd112a585dc/pkg/imgpkg/bundle/contents.go#L89-L92 (was already prepped for recursive bundles with a allowInnerBundles bool.

Reply:
Hmmm i'm not following / missing a nuance here...

This check happens during a `copy`. Specifically they are trying to copy a bundle that contains nested bundles, but they haven't provided the feature flag. In this case we want to provide a meaningful error message to the user, teaching them about this flag.

From the issue, it is implementing `Scenario 3: Fail without --experimental-recursive-bundle on recursive bundle bundles using a bundle lock file`

Reply:
if we remove this function before release then this doesnt matter.

in case we keep this function in the next release then this affects performance of regular users that use imgpkg because we would be iterating over each image and fetching its config to check if something is a bundle. this flag was made to ensure that we do not "ship" recursive bundle code too early afaik, not with an intent that we would be encouraging ppl to use it (since it's not ready for consumption). (in other words this code favors experimental users instead of regular users).

(code i linked above prevents users from making bundles with nested bundles if flag is not used, hence we know that user knows about the flag).

Reply:
> if we remove this function before release then this doesnt matter.
True.

I am going to assume we ship a release *with* the feature flag still in... Because, you know, an urgent security fix had to be applied...

In that case, the only risk I can think of is, if someone tries to copy a bundle (that contains a nested bundle), and they do *not* specify the feature flag (because to err is human), then the copy command will succeed (Even though the nested bundles haven't been copied)

Eh, The above concern is low risk IMO. So I'll remove the check and update the tests too to reflect this change in behavior.